addons/calendar/models/res_partner.py

```diff
@@ -12,24 +12,33 @@ class Partner(models.Model):
     calendar_last_notif_ack = fields.Datetime(
         'Last notification marked as read from base Calendar', default=fields.Datetime.now)
 
-    def get_attendee_detail(self, meeting_id):
-        """ Return a list of tuple (id, name, status)
-            Used by base_calendar.js : Many2ManyAttendee
+    def get_attendee_detail(self, meeting_ids):
+        """ Return a list of dict with the attendees details
+            Used by:
+                - base_calendar.js : Many2ManyAttendee
+                - calendar_model.js (calendar.CalendarModel)
         """
-        datas = []
-        meeting = None
-        if meeting_id:
-            meeting = self.env['calendar.event'].browse(meeting_id)
-
-        for partner in self:
-            data = partner.name_get()[0]
-            data = [data[0], data[1], False, partner.color]
-            if meeting:
-                for attendee in meeting.attendee_ids:
-                    if attendee.partner_id.id == partner.id:
-                        data[2] = attendee.state
-            datas.append(data)
-        return datas
+        attendees_details = []
+        meetings = self.env['calendar.event'].browse(meeting_ids)
+
+        for meeting in meetings:
+            for partner in self:
+                partner_info = partner.name_get()[0]
+                meeting_details = dict({
+                    'id': partner_info[0],
+                    'name': partner_info[1],
+                    'color': partner.color,
+                })
```

Comment:
this is confusing as you set a dict that could not be used if partner is not in any meetings.
I propose this rewrite:
```
for meeting in meetings:
    for partner in self:
        partner_info = partner.name_get()[0]
        for attendee in meeting.attendee_ids.filtered(lambda att: att.partner_id.id == partner.id):
            attendee_is_organizer = self.env.user == meeting.user_id and attendee.partner_id == self.env.user.partner_id
            attendees_details.append({
                'id': partner_info[0],
                'name': partner_info[1],
                'color': partner.color,
                'status': attendee.state,
                'event_id': meeting.id,
                'attendee_id': attendee.id,
                'is_alone': meeting.is_organizer_alone and attendee_is_organizer,
            })
```
and to avoid the triple loop we could even do:
```
meetings_attendees = meetings.mapped('attendee_ids')
for partner in self:
    for attendee in meetings_attendees.filtered(lambda att: att.partner_id.id == partner.id):
        ...
```
and use the attendee to get the the meeting_id and meeting's organizer.
What do you think ?